src/stores/AccountController.ts

```diff
@@ -229,67 +233,81 @@ export const updatePersonalShelves = async () => {
   });
 };
 
-export const updateConsents = async (customerConsents: CustomerConsent[]) => {
-  return await useLoginContext(async ({ cleengSandbox, customerId, auth: { jwt } }) => {
-    const response = await cleengAccountService.updateCustomerConsents(
-      {
-        id: customerId,
+export const updateConsents = async (customerConsents: CustomerConsent[]): Promise<ServiceResponse<never>> => {
+  return await useAccountContext(async ({ customer, auth: { jwt, refreshToken } }) => {
+    return await withAccountService(async ({ accountService, config }) => {
+      useAccountStore.setState({ loading: true });
+
+      const response = await accountService.updateCustomerConsents({
+        jwt,
+        config,
+        customer,
         consents: customerConsents,
-      },
-      cleengSandbox,
-      jwt,
-    );
+      });
 
-    await getCustomerConsents();
+      // if user does not have refresh token (InPlayer user) the response data is user object
+      if (response?.responseData && !refreshToken) {
```

Comment:
This is a kind of messy execution path.  There are multiple callback helpers nested both here and 4 levels deep in the call to getCustomerConsents below.  You're also passing data between functions using the store, which is valid, but hard to follow.

I propose instead we modify the cleeng updateCustomerConsents to include the getCustomerConsents call like what you did for the other calls in the register flow.  Then the generic updateCustomerConsents can just return the customer consents as well.

Reply:
So inside of updateConsents we have:

- useAccountContext
  - withAccountService
    - accountService.updateCustomerConsents
    - useAccountStore.setState(user)
    - getCustomerConsents
      - useAccountContext which reads the user set 2 lines above
        - withAccountService
          - accountService.getCustomerConsents
          - useAccountStore.setState(customerConsents)
          
So we're calling the wrappers multiple times, which won't be terrible for performance, but does make it hard to follow exactly what is happening, especially with tricks like `setState(user)` updating the store which is then read with the second `useAccountContext` which passed it to `getCustomerConsents` which is a dummy API call that just grabs data off of the user object that we've passed around.  It's even more complex because the setState(user) only happens for the inplayer service based off checking the refresh token.

What is a better flow is to do this:
- accountService.updateCustomerConsents returns `customerConsents`
  - Inplayer service can already do this.  We don't need to include the other `user` data.
  - Cleeng service will just also call `getCustomerConsents` inside of the `updateCustomerConsents` method
- We will also have to update `getUser` to return user and customerConsents
  - Again, inplayer service can already do this
  - Cleeng can again just add a call to `getCustomerConsents` inside of the `getUser` method
- The above change will propagate customerConsents to also be able to be returned from `register` and `login`
- We can add customerConsents to the setState in afterLogin and remove getCustomerConsents from afterLogin
- getCustomerConsents can be removed from the accountController
- getCustomerConsents dummy method can be removed from inplayer and cleeng will only be private (not exported) for cleeng

Reply:
Yes, it requires refactoring but it makes sense.